src/Common.OData.ApiExplorer/AspNet.OData/DefaultModelTypeBuilder.cs

```diff
@@ -171,14 +171,15 @@ public Type NewStructuredType( IEdmStructuredType structuredType, Type clrType,
         }
 
         /// <inheritdoc />
-        public Type NewActionParameters( IServiceProvider services, IEdmAction action, ApiVersion apiVersion )
+        public Type NewActionParameters( IServiceProvider services, IEdmAction action, ApiVersion apiVersion, string controllerName )
         {
             Arg.NotNull( services, nameof( services ) );
             Arg.NotNull( action, nameof( action ) );
             Arg.NotNull( apiVersion, nameof( apiVersion ) );
+            Arg.NotNull( controllerName, nameof(controllerName) );
             Contract.Ensures( Contract.Result<Type>() != null );
 
-            var name = action.FullName() + "Parameters";
+            var name = controllerName + "." + action.FullName() + "Parameters";
```

Comment:
I can't remember now, but does this handle whether different versions of the action exist by API version? I _think_ it's still possible for the `controllerName` and `action.FullName()` to be the same. Simply calling to `apiVersion.ToString()` and adding that to the name probably won't work because it will produce an invalid name. Maybe `apiVersion.GetHashCode().ToString()` somewhere in the mix? Does this even matter? Isn't it handled by the types being generated in separate assemblies per API version? I guess is just covering the case where two different controllers define actions with the same name? Just double-checking.

Reply:
The problem I was having is duplicate type names within the same assembly. This is only per api version since a new assembly is created per api version.

This covers the case where two actions with the same name would be defined in the same api version in different controllers.